### ISSUE TITLE
crash with notifications enabled

### ISSUE THREAD

**Issue by nick75e** [elboulangero/goodvibes#79]
Hi,

When I enable the notifications, Goodvibes crashes after some time.
```
$ goodvibes --status-icon
WARN 00:52:30 [GLib-GIO] unable to send notifications through org.freedesktop.Notifications: Le délai d’attente est dépassé
Erreur de segmentation (core dumped)
```
_"Le délai d’attente est dépassé"_ = timed out
_"Erreur de segmentation"_ = segmentation fault

Linux Mint Cinnamon 18.3, Goodvibes 0.3.4

**Comment by elboulangero** [comment]
Hey @nick75e,

can you reproduce that easily ? Is there a particular radio station that you listen to when you get this crash ? That would be a good starting point for me. The stations I listen to don't stream metadata, so even when I enable notifications, there's nothing to notify.

Also, from which repo did you install Goodvibes ?

Thanks !

**Comment by nick75e** [comment]
It seems random but happens quicker when switching between stations (within a minute). Here are some stations with metadata:
```
http://radiofg.impek.com/fg
http://radiofg.impek.com/fge
http://broadcast.infomaniak.net/start-adofm-high.mp3
http://lamaxradio.ice.infomaniak.ch/la-max-radio-128.mp3
```
I installed Goodvibes from your repository `ppa:elboulangero/goodvibes`

**Comment by elboulangero** [comment]
I can't reproduce that issue on GNOME.

What could happen is that the Cinnamon notification daemon crashes upon receiving a notification from Goodvibes. Do you think you can verify that ? I don't know what is the process in charge of showing notifications on Cinnamon, do you know what process it is ?

If you know what process shows the notifications, then you can look at its pid (process id) with `ps | grep PROCESS_NAME` (replace PROCESS_NAME by the name of the process). Play a bit with Goodvibes until it crashes, and then look again at the process with `ps`. If the process id changed, then there's a good chance that it crashed and was restarted.

Another, maybe much simpler way of debugging this issue: launch a shell, and type `sudo journalctl -f` to monitor the system logs. Then play with Goodvibes until it crashes. Maybe you will see something else happening in the system logs, like the notification daemon crashing. If there's anything to see, it's very likely that it will happen much before Goodvibes crash, something around 30 seconds before.

**Comment by nick75e** [comment]
this is the output of journalctl -f
1st time
```
mars 07 18:22:20 my_computer kernel: goodvibes[2679]: segfault at 10000000005 ip 00007f16a4f0ca8a sp 00007fffa3300990 error 4 in libglib-2.0.so.0.4800.2[7f16a4ea6000+10f000]
```
2nd time
```
mars 07 18:29:44 my_computer kernel: goodvibes[4968]: segfault at 40 ip 00007f8241f25b08 sp 00007ffc6f192e70 error 4 in libglib-2.0.so.0.4800.2[7f8241ebe000+10f000]
```

Nothing else seems to crash, I still get notifications before and after Goodvibes crashes.
Hope it helps.

**Comment by elboulangero** [comment]
Ok, doesn't really help. One last question: did it work at some point ? Do you remember of a Ubuntu version where it worked before ?

**Comment by nick75e** [comment]
Sorry for the delay.

No, I don't. I was just messing about with the options and noticed Goodvibes would randomly crash.
:-/

**Comment by elboulangero** [comment]
Hey, good news I've been able to reproduce that in a VM, I'm working on a fix.

**Comment by elboulangero** [comment]
Closed in #79, I will do a release soon
